Source/TestDafny/MultiBackendTest.cs

```diff
@@ -339,15 +357,70 @@ int ExpectedExitCode
       dafnyArgs = dafnyArgs.Concat(new[] { "--include-runtime:false", "--input", runtimePath });
     }
 
-    var (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    int exitCode;
+    string outputString;
+    string error;
+    try {
+      (exitCode, outputString, error) = await RunDafny(options.DafnyCliPath, dafnyArgs);
+    } catch (Exception e) {
+      (exitCode, outputString, error) = (3, e.ToString(), e.ToString());
+    }
+
     var compilationOutputPrior = new Regex("\r?\nDafny program verifier[^\r\n]*\r?\n").Match(outputString);
     if (compilationOutputPrior.Success) {
       outputString = outputString.Remove(0, compilationOutputPrior.Index + compilationOutputPrior.Length);
     }
 
+    var diffMessage = exitCode == 0 ? AssertWithDiff.GetDiffMessage(expectedOutput, outputString) : null;
+    if (checkFile == null && exitCode != 0) {
+      if (Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE") == "true") {
+        if ((Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_ROOT_DIR") ?? "") == "") {
+          await output.WriteLineAsync(
+            "DAFNY_INTEGRATION_TESTS_UPDATE_TARGET_EXPECT_FILE is true but DAFNY_INTEGRATION_TESTS_ROOT_DIR is not set");
+        } else {
+          // Ensure we don't accidentally do this for every compiler.
+          var sourcePath = Path.Join(Environment.GetEnvironmentVariable("DAFNY_INTEGRATION_TESTS_ROOT_DIR"),
+            CheckFileForBackend(options, backend));
+          var contentCheck = (outputString + "\n" + error).Trim();
+          var checkOutput = "";
+          if (contentCheck == "") {
+            checkOutput = string.Join("\n", expectedOutput.Split("\n").Select(line => "// CHECK-NOT: " + line));
+          } else {
+            checkOutput = string.Join("\n",
+              contentCheck.Split("\n").Select(line => "// CHECK-L: " + line));
+          }
+
+          if (!File.Exists(sourcePath)) {
+            await File.WriteAllTextAsync(sourcePath, checkOutput);
+          } else {
+            await output.WriteLineAsync($"Check file existed but was not imported yet. Ignoring.");
```

Comment:
Not following, what do you mean by "imported" here?

Reply:
During development, it often happens that I update an, auto-generated `.check` file in `IntegrationTests/TestFiles` folder to make a test pass. However, the test runner does not automatically copy this new file to the `/bin` folder if I don't change anything else like a `.dfy` file or a `.cs` file. Hence, the second time I run a test, previously it was overwriting my updated `.check` file because it thought it did not exist in the `/bin` folder. This line ensures that this case is detected and ignored, so that I just remember to force rebuild the solution or copy the file manually to the `/bin` folder;

Reply:
Ah okay. I would just expand the output line a bit to explain that, and/or a comment above the source line.